src/modules/status.rs

```diff
@@ -55,22 +55,33 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     }
 
     let segment_format = config.pipestatus_segment_format.unwrap_or(config.format);
+    let segment_format_with_separator = [segment_format, config.pipestatus_separator].join("");
 
-    // Create pipestatus string
     let pipestatus = match pipestatus_status {
         PipeStatusStatus::Pipe(pipestatus) => pipestatus
             .iter()
-            .map(
-                |ec| match format_exit_code(ec.as_str(), segment_format, None, &config, context) {
+            .enumerate()
+            .map(|(i, ec)| {
+                match format_exit_code(
+                    ec.as_str(),
+                    if i == pipestatus.len() - 1 {
+                        segment_format
+                    } else {
+                        &segment_format_with_separator
+                    },
+                    None,
+                    &config,
+                    context,
+                ) {
                     Ok(segments) => segments
                         .into_iter()
                         .map(|s| s.to_string())
                         .collect::<String>(),
                     Err(_) => "".to_string(),
-                },
-            )
+                }
+            })
             .collect::<Vec<String>>()
-            .join(config.pipestatus_separator),
+            .join(""),
```

Comment:
This would look a bit nicer:
```rust
            .enumerate()
            .filter_map(|(i, ec)| {
                format_exit_code(
                    ec.as_str(),
                    if i == pipestatus.len() - 1 {
                        segment_format
                    } else {
                        &segment_format_with_separator
                    },
                    None,
                    &config,
                    context,
                )
                .ok()
                .map(|segments| segments.into_iter().map(|s| s.to_string()))
            })
            .flatten()
            .collect::<String>(),
```